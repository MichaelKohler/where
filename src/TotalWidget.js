import React from 'react';
import PropTypes from 'prop-types';
import './scss/widget.scss';

class TotalWidget extends React.Component {
  render() {
    let sectionClasses = 'card vertical';
    if (this.props.color) {
      sectionClasses += ` ${this.props.color}`;
    }

    return (
      <section className={sectionClasses}>
        <div className="number">
          <h2>{this.props.totals.trips}</h2>
          <p>total trips</p>
        </div>
        <div className="number">
          <h2>{this.props.totals.flights}</h2>
          <p>total flights</p>
        </div>
        <div className="number">
          <h2>{this.props.totals.countries}</h2>
          <p>total countries</p>
        </div>
      </section>
    );
  }
}

TotalWidget.defaultProps = {
  totals: {
    trips: 0,
    flights: 0,
    countries: 0
  },
  color: ''
};

TotalWidget.propTypes = {
  totals: PropTypes.shape({
    trips: PropTypes.number,
    flights: PropTypes.number,
    countries: PropTypes.number,
  }),
  color: PropTypes.string,
};

export default TotalWidget;
