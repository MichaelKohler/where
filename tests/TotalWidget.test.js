import React from 'react';
import { render } from '@testing-library/react';

import TotalWidget from '../src/TotalWidget';

test('should correctly shallow render TotalWidget', () => {
  const totals = {
    trips: 200,
    countries: 20,
    uniqueDestinations: 15,
  };
  const { container } = render(<TotalWidget totals={totals} color="blue" />);
  expect(container.firstChild).toMatchSnapshot();
});
