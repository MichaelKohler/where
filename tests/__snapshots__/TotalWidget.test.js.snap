// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly shallow render TotalWidget 1`] = `
<section
  class="card vertical blue"
>
  <div
    class="number"
  >
    <h2>
      200
    </h2>
    <p>
      total trips
    </p>
  </div>
  <div
    class="number"
  >
    <h2>
      15
    </h2>
    <p>
      unique cities
    </p>
  </div>
  <div
    class="number"
  >
    <h2>
      20
    </h2>
    <p>
      total countries
    </p>
  </div>
</section>
`;
